src/pool/srv_pool.c

```diff
@@ -7088,6 +7239,13 @@ pool_svc_update_map_internal(struct pool_svc *svc, unsigned int opc,
 			DL_INFO(rc, DF_UUID": failed to schedule RF check", DP_UUID(svc->ps_uuid));
 	}
 
+	rc = pool_svc_update_map_metrics(svc->ps_uuid, map,
+					 svc->ps_pool->sp_metrics[DAOS_POOL_MODULE]);
+	if (rc != 0) {
+		DL_ERROR(rc, DF_UUID ": failed to update pool metrics", DP_UUID(svc->ps_uuid));
+		rc = 0; /* not fatal */
```

Comment:
Should the same be true in the step_up code, that if the call to pool_svc_update_map_metrics() fails there, it would be considered non-fatal?

Reply:
I have to admit to a bit of dithering on this due to my unfamiliarity with this area of the codebase. In other areas of the code, there is no defensive programming around access to the per-module metrics. I guess maybe the idea is that it's a "can't happen unless things are really messed up" kind of situation and so there's no point? Seems like I should probably pick one side or the other though, I agree.

Reply:
At any rate... I just adjusted the other call to `pool_svc_update_map_metrics()` to be non-fatal as well. Also changed the ERRORs to WARNs.